docker/service.go

```diff
@@ -344,21 +358,58 @@ func (s *Service) collectContainersAndDo(ctx context.Context, action func(*Conta
 }
 
 func (s *Service) eachContainer(ctx context.Context, containers []*Container, action func(*Container) error) error {
-
 	tasks := utils.InParallel{}
 	for _, container := range containers {
-		task := func(container *Container) func() error {
-			return func() error {
-				return action(container)
-			}
-		}(container)
+		task := taskFunc(container, action)
+		tasks.Add(task)
+	}
+	return tasks.Wait()
+}
+
+func (s *Service) getFilters() filters.Args {
+	filter := filters.NewArgs()
+	filter.Add("label", fmt.Sprintf("%s=%s", labels.PROJECT, s.project.Name))
+	filter.Add("label", fmt.Sprintf("%s=%s", labels.SERVICE, s.name))
+	return filter
+}
 
+func (s *Service) watchContainers(ctx context.Context, initialContainers []*Container, action func(*Container) error, dieAction func(*Container) error) error {
```

Comment:
@kunalkushwaha hum, for `--abort-on-container-exit` you might only need to listen to `die` event of containers filtered by the project/service (what `watchContainers` do).

But that's a good point, I'll see if I find a good way of handling that :+1: